java/openmldb-jdbc/src/main/java/com/_4paradigm/openmldb/sdk/impl/InsertPreparedStatementImpl.java

```diff
@@ -63,17 +66,25 @@ public InsertPreparedStatementImpl(String db, String sql, SQLRouter router) thro
         this.currentSchema = tempRow.GetSchema();
         VectorUint32 idxes = tempRow.GetHoleIdx();
 
+        // In stmt order, if no columns in stmt, in schema order
+        // We'll sort it to schema order later, so needs the map <real_schema_idx, current_data_idx>
+        schemaIdxes = new ArrayList<>(idxes.size());
+        for (int i = 0; i < idxes.size(); i++) {
+            schemaIdxes.add(new Pair<>(idxes.get(i), i));
+        }
+
         currentDatas = new ArrayList<>(idxes.size());
         currentDatasType = new ArrayList<>(idxes.size());
         hasSet = new ArrayList<>(idxes.size());
-        scehmaIdxs = new ArrayList<>(idxes.size());
-        for (int i = 0; i < idxes.size(); i++) {
-            long idx = idxes.get(i);
+        // CurrentDatas and Type order is consistent with insert stmt. We'll do appending in schema order when build
+        // row.
+        for (Pair<Long, Integer> pair : schemaIdxes) {
```

Comment:
it can be merged into one loop with the above `for` statement ?

Reply:
fixed